### PR TITLE
pino implemented and worker job update feature add

### PR DESCRIPTION
**Pull request by bhumilsarvaiya**
- **Pino Logger** is now implemented. Console logs can now be toggled by making single change in config file.

- **update service** to update worker jobs added.

- **_'created'_** status can now be retained while creating new job by setting the parameter options>manualStatus.

- job creation bug fix. i.e. 'options' passed will not be padded to data field of the job. Previously, the bug was occuring if jobs were created as : 

```javascript
{
    ...,
    "options" : {
        //this options were not padded
    },
    "jobs":[
        {
	    ...[data fields],
	    "options": {
	        //options passed here were getting padded into the data field of job
	    }
	}
    ],
    ...
}
```
**OR**

```javascript
{
   ...,
    "options" : {
    	//options passed here were getting padded into the data field of job
    },
    ...[data fields]
}
```

### CODE REVIEW

File: config/default.json [hunk 1]
```diff
@@ -29,10 +33,19 @@
     "port": 3000,
     "getJobModuleApiURL": "http://localhost:3000/job-module/"
   },
+  "waitingThreshold": 0.40,
+  "increaseWorker": 10,
+  "maxWorker": 10,
```
Comment (chiragob) [inline config/default.json]: "waitingThreshold": 0.40,
"increaseWorker": 10,
"maxWorker": 10,
 add to symmetricWorker option

File: job.js [hunk 2]
```diff
@@ -357,6 +377,57 @@ module.exports = function job (options) {
...
+        }).catch(err => reject(err))
+      } catch (err) {
+        reject(err)
+      }
+    })
+  }
+
 }
 
 let customError = function (err, errorCode) {
```
Comment (chiragob) [inline job.js:433]: set invalid parameter error key
date enable error

File: symmetricWorker.js [hunk 3]
```diff
@@ -4,76 +4,82 @@ var _ = require('underscore')
 var app = require('express')()
 var server = require('http').Server(app)
 var io = require('socket.io')(server)
+const rethink = require('rethinkdb')
+const defaultConnection = config.get('defaultConnection')
+const symmetricWorker = config.get('symmetricWorker')
+const pino = require('pino')
+
+const waitingThreshold = config.get('waitingThreshold')
```
Comment (chiragob) [inline symmetricWorker.js]: add to symmetricWorker option

**Review by chiragob — CHANGES_REQUESTED** [review]

File: symmetricWorker.js (hunk 3 above)
Comment (bhumilsarvaiya) [reply symmetricWorker.js]: requested changes are addressed in new commit ***'requested changes in review'*** .
due to changes in config, necessary changes are also made in this file.

File: job.js (hunk 2 above)
Comment (bhumilsarvaiya) [reply job.js:433]: requested changes are addressed in new commit ***'requested changes in review'*** .
**err.name** is now used to identify the error.

File: config/default.json (hunk 1 above)
Comment (bhumilsarvaiya) [reply config/default.json]: requested changes are addressed in new commit ***'requested changes in review'*** .
the required fields now moved to **symmetricWorker** option

**Review by chiragob — APPROVED** [review]

**Pull request merged**